sql: TestShowRangesMultipleStores failed

sql.TestShowRangesMultipleStores [failed](https://tanzanite.cluster.engflow.com/invocations/default/17ca139c-cd9b-4616-816d-0c1bdd1afaec?testReportRun=1&testReportShard=12&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [1b0a374fd2a101cebcfb24cff4b3b57795ad1df6](https://github.com/cockroachdb/cockroach/commits/1b0a374fd2a101cebcfb24cff4b3b57795ad1df6):


```
=== RUN   TestShowRangesMultipleStores
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestShowRangesMultipleStores761670925
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    show_ranges_test.go:147: -- test log scope end --
test logs left over in: outputs.zip/logTestShowRangesMultipleStores761670925
--- FAIL: TestShowRangesMultipleStores (123.11s)
=== RUN   TestShowRangesMultipleStores/SHOW_RANGES_FROM_DATABASE_system_WITH_DETAILS
    sql_runner.go:121: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:134
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:365
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/show_ranges_test.go:137
        	Error:      	Received unexpected error:
        	            	query '
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:76
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:218
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:82
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:134
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:365
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestShowRangesMultipleStores.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/show_ranges_test.go:137
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1689
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) query '
        	            	  | SELECT DISTINCT
        	            	  | 		(
        	            	  | 		array_position(replica_localities, lease_holder_locality)
        	            	  | 		= array_position(replicas, lease_holder)
        	            	  | 		)
        	            	  | 	FROM [SHOW RANGES FROM DATABASE system WITH DETAILS]': expected:
        	            	  | true
        	            	  |
        	            	  | got:
        	            	  | true
        	            	  | NULL
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    --- FAIL: TestShowRangesMultipleStores/SHOW_RANGES_FROM_DATABASE_system_WITH_DETAILS (60.75s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=12</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowRangesMultipleStores.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-42088

sql.TestShowRangesMultipleStores [failed](https://tanzanite.cluster.engflow.com/invocations/default/0ba9a8d8-3d04-455d-b066-a74a801783e2?testReportRun=2&testReportShard=12&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [31794d8d89e3e117e1654cb13ad109da04de5445](https://github.com/cockroachdb/cockroach/commits/31794d8d89e3e117e1654cb13ad109da04de5445):


```
=== RUN   TestShowRangesMultipleStores
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestShowRangesMultipleStores733170128
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    show_ranges_test.go:147: -- test log scope end --
test logs left over in: outputs.zip/logTestShowRangesMultipleStores733170128
--- FAIL: TestShowRangesMultipleStores (134.07s)
=== RUN   TestShowRangesMultipleStores/SHOW_RANGES_FROM_DATABASE_system_WITH_DETAILS
    sql_runner.go:121: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:134
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:365
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/show_ranges_test.go:137
        	Error:      	Received unexpected error:
        	            	query '
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:76
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:218
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:82
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:134
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:365
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestShowRangesMultipleStores.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/show_ranges_test.go:137
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1689
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) query '
        	            	  | SELECT DISTINCT
        	            	  | 		(
        	            	  | 		array_position(replica_localities, lease_holder_locality)
        	            	  | 		= array_position(replicas, lease_holder)
        	            	  | 		)
        	            	  | 	FROM [SHOW RANGES FROM DATABASE system WITH DETAILS]': expected:
        	            	  | true
        	            	  |
        	            	  | got:
        	            	  | true
        	            	  | NULL
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    --- FAIL: TestShowRangesMultipleStores/SHOW_RANGES_FROM_DATABASE_system_WITH_DETAILS (61.02s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=12</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowRangesMultipleStores.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestShowRangesMultipleStores [failed](https://tanzanite.cluster.engflow.com/invocations/default/681bf7cd-f8c6-43f4-b47b-b7769019fea7?testReportRun=1&testReportShard=12&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [3f8226eb7f09eb93c26dda4cd47e023a8f56ea23](https://github.com/cockroachdb/cockroach/commits/3f8226eb7f09eb93c26dda4cd47e023a8f56ea23):


```
=== RUN   TestShowRangesMultipleStores
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestShowRangesMultipleStores406550713
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    show_ranges_test.go:147: -- test log scope end --
test logs left over in: outputs.zip/logTestShowRangesMultipleStores406550713
--- FAIL: TestShowRangesMultipleStores (120.17s)
=== RUN   TestShowRangesMultipleStores/SHOW_RANGES_FROM_DATABASE_system_WITH_DETAILS
    sql_runner.go:121: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:134
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:365
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/show_ranges_test.go:137
        	Error:      	Received unexpected error:
        	            	query '
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:76
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:218
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:82
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:134
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:365
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestShowRangesMultipleStores.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/show_ranges_test.go:137
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1689
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) query '
        	            	  | SELECT DISTINCT
        	            	  | 		(
        	            	  | 		array_position(replica_localities, lease_holder_locality)
        	            	  | 		= array_position(replicas, lease_holder)
        	            	  | 		)
        	            	  | 	FROM [SHOW RANGES FROM DATABASE system WITH DETAILS]': expected:
        	            	  | true
        	            	  |
        	            	  | got:
        	            	  | true
        	            	  | NULL
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    --- FAIL: TestShowRangesMultipleStores/SHOW_RANGES_FROM_DATABASE_system_WITH_DETAILS (55.83s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=12</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #130556 sql: TestShowRangesMultipleStores failed [C-test-failure O-robot T-sql-queries branch-release-24.1 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowRangesMultipleStores.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestShowRangesMultipleStores [failed](https://tanzanite.cluster.engflow.com/invocations/default/34138952-25a8-4a2c-9b09-114f10054b32?testReportRun=3&testReportShard=12&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [83589fb87caa92fb42e83994f1691978f37e4cbb](https://github.com/cockroachdb/cockroach/commits/83589fb87caa92fb42e83994f1691978f37e4cbb):


```
=== RUN   TestShowRangesMultipleStores
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestShowRangesMultipleStores3690225305
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    show_ranges_test.go:147: -- test log scope end --
test logs left over in: outputs.zip/logTestShowRangesMultipleStores3690225305
--- FAIL: TestShowRangesMultipleStores (113.51s)
=== RUN   TestShowRangesMultipleStores/SHOW_RANGES_FROM_DATABASE_system_WITH_DETAILS
    sql_runner.go:121: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:134
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:365
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/show_ranges_test.go:137
        	Error:      	Received unexpected error:
        	            	query '
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:368
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:76
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:218
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:82
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:134
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:365
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestShowRangesMultipleStores.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/show_ranges_test.go:137
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1689
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) query '
        	            	  | SELECT DISTINCT
        	            	  | 		(
        	            	  | 		array_position(replica_localities, lease_holder_locality)
        	            	  | 		= array_position(replicas, lease_holder)
        	            	  | 		)
        	            	  | 	FROM [SHOW RANGES FROM DATABASE system WITH DETAILS]': expected:
        	            	  | true
        	            	  |
        	            	  | got:
        	            	  | true
        	            	  | NULL
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    --- FAIL: TestShowRangesMultipleStores/SHOW_RANGES_FROM_DATABASE_system_WITH_DETAILS (48.60s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=12</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowRangesMultipleStores.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestShowRangesMultipleStores [failed](https://tanzanite.cluster.engflow.com/invocations/default/2b552d05-63b2-424a-aeb6-f5bf2763a694?testReportRun=2&testReportShard=12&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [8c52619671966cf15055f8b2e271ea82b59440c5](https://github.com/cockroachdb/cockroach/commits/8c52619671966cf15055f8b2e271ea82b59440c5):


```
=== RUN   TestShowRangesMultipleStores
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestShowRangesMultipleStores3725865541
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    show_ranges_test.go:147: -- test log scope end --
test logs left over in: outputs.zip/logTestShowRangesMultipleStores3725865541
--- FAIL: TestShowRangesMultipleStores (134.17s)
=== RUN   TestShowRangesMultipleStores/SHOW_RANGES_FROM_DATABASE_system_WITH_DETAILS
    sql_runner.go:121: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:134
        	            				github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:377
        	            				github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/show_ranges_test.go:137
        	Error:      	Received unexpected error:
        	            	query '
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:380
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:76
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	github.com/cockroachdb/cockroach/pkg/util/retry/retry.go:218
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:82
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:134
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:377
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestShowRangesMultipleStores.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/show_ranges_test.go:137
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1689
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) query '
        	            	  | SELECT DISTINCT
        	            	  | 		(
        	            	  | 		array_position(replica_localities, lease_holder_locality)
        	            	  | 		= array_position(replicas, lease_holder)
        	            	  | 		)
        	            	  | 	FROM [SHOW RANGES FROM DATABASE system WITH DETAILS]': expected:
        	            	  | true
        	            	  |
        	            	  | got:
        	            	  | true
        	            	  | NULL
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    --- FAIL: TestShowRangesMultipleStores/SHOW_RANGES_FROM_DATABASE_system_WITH_DETAILS (56.31s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=12</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowRangesMultipleStores.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestShowRangesMultipleStores [failed](https://tanzanite.cluster.engflow.com/invocations/default/94664c35-10a6-4f48-a11b-30ecf1984958?testReportRun=4&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [9c3acd84e2fc639328b27a6cacd0e3893f1f90bb](https://github.com/cockroachdb/cockroach/commits/9c3acd84e2fc639328b27a6cacd0e3893f1f90bb):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x185a991]

```
Stack: 

```
goroutine 187810 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc00cff3868?, {0x8b9c428, 0xc009a4e600})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:226 +0x65
panic({0x66c1600?, 0xce02b40?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc00cff3988?, {0x8b9c428, 0xc009a4e840})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:226 +0x65
panic({0x66c1600?, 0xce02b40?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotQueue).Admit(0x0, {0x8b9c428, 0xc009a4ee40}, 0x563)
	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:165 +0x71
github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotPacer).Pace(0xc00cff4768, {0x8b9c428?, 0xc009a4ee40?}, 0xc008199340?, 0x28?)
	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:292 +0x6d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*kvBatchSnapshotStrategy).Receive(_, {_, _}, _, {_, _}, {{0xe, 0x2, 0xc0072c4380, 0xc0072c43f0, ...}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:918 +0x1f9f
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).receiveSnapshot(0xc013c8a008, {0x8b9c428, 0xc009a4e840}, 0xc007d9d448, {0x7fab219213b0, 0xc013e4f220})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:1693 +0x76b
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot.func1({0x8b9c428, 0xc009a4e840})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:209 +0x85
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc002abc120, {0x8b9c428, 0xc009a4e840}, {0x7283f3b, 0x1e}, 0xc00bca5280)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349 +0x15d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot(0xc013c8a008, {0x8b9c3f0?, 0xc013810ff0?}, 0xc007d9d448, {0x7fab21921388, 0xc013e4f220})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:206 +0xf6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*RaftTransport).RaftSnapshot(0xc00637b180, {0x8bea810, 0xc013e4f220})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/raft_transport.go:638 +0x32d
github.com/cockroachdb/cockroach/pkg/kv/kvserver._MultiRaft_RaftSnapshot_Handler({0x70b4b80?, 0xc00637b180}, {0x8bd3d18, 0xc01b1ba300})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvserver/storage_services.pb.go:267 +0xd8
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.StreamServerInterceptor.func13({0x70b4b80, 0xc00637b180}, {0x8bd4f58, 0xc00c96e3c0}, 0xc006811368, 0x76ad660)
	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:160 +0x35b
google.golang.org/grpc.getChainStreamHandler.func1({0x70b4b80, 0xc00637b180}, {0x8bd4f58, 0xc00c96e3c0})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4({0x70b4b80, 0xc00637b180}, {0x8bd4f58, 0xc00c96e3c0}, 0xc006811368?, 0xc0052fd7c0)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:174 +0x6c
google.golang.org/grpc.getChainStreamHandler.func1({0x70b4b80, 0xc00637b180}, {0x8bd4f58, 0xc00c96e3c0})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor({0xc00e61e000?, {{0x470445?}, {0x8bf5f20?, 0xc00cc5db60?}}}, {0x70b4b80, 0xc00637b180}, {0x8bd4f58, 0xc00c96e3c0}, 0xc006811368, 0xc0052fd780)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:152 +0x384
google.golang.org/grpc.getChainStreamHandler.func1({0x70b4b80, 0xc00637b180}, {0x8bd4f58, 0xc00c96e3c0})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1({0x7fab2398bc98?, 0x8b9c428?})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:141 +0x27
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc002abc120, {0x8b9c428, 0xc009a4e600}, {0xc00a2eeff0, 0x29}, 0xc005353aa0)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349 +0x15d
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2({0x70b4b80, 0xc00637b180}, {0x8bd4f58, 0xc00c96e3c0}, 0xc006811368, 0xc0052fd740)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:140 +0xdf
google.golang.org/grpc.NewServer.chainStreamServerInterceptors.chainStreamInterceptors.func2({0x70b4b80, 0xc00637b180}, {0x8bd4f58, 0xc00c96e3c0}, 0xc006811368, 0xc013e4f210?)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1472 +0x85
google.golang.org/grpc.(*Server).processStreamingRPC(0xc00a6ec000, {0x8bf6880, 0xc00bc029c0}, 0xc012cf97a0, 0xc015b5c360, 0xceffca0, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1636 +0x1247
google.golang.org/grpc.(*Server).handleStream(0xc00a6ec000, {0x8bf6880, 0xc00bc029c0}, 0xc012cf97a0, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1716 +0x999
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:947 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 187776
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:958 +0x136
```
<details><summary>Log preceding fatal error</summary>
<p>

```
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotQueue).Admit
*   | 	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:165
*   | github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotPacer).Pace
*   | 	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:292
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*kvBatchSnapshotStrategy).Receive
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:918
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).receiveSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:1693
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot.func1
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:209
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:206
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*RaftTransport).RaftSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/raft_transport.go:638
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver._MultiRaft_RaftSnapshot_Handler
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvserver/storage_services.pb.go:267
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.StreamServerInterceptor.func13
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:160
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:174
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:152
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:141
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:140
*   | google.golang.org/grpc.NewServer.chainStreamServerInterceptors.chainStreamInterceptors.func2
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1472
*   | google.golang.org/grpc.(*Server).processStreamingRPC
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1636
*   | google.golang.org/grpc.(*Server).handleStream
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1716
*   | google.golang.org/grpc.(*Server).serveStreams.func1.1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:947
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=4</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowRangesMultipleStores.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestShowRangesMultipleStores [failed](https://tanzanite.cluster.engflow.com/invocations/default/7cafe118-d052-4671-8963-4873d7504c6e?testReportRun=2&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [472ea07a5232c98536293d13bb46cca59f9f2cd0](https://github.com/cockroachdb/cockroach/commits/472ea07a5232c98536293d13bb46cca59f9f2cd0):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x26fed37]

```
Stack: 

```
goroutine 103070 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc008b14d00?, {0xe87b668, 0xc006c4bc80})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:226 +0x74
panic({0xa674ca0?, 0x12ee6b70?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc008b14e38?, {0xe87b668, 0xc006c4bef0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:226 +0x74
panic({0xa674ca0?, 0x12ee6b70?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotQueue).Admit(0x0, {0xe87b668, 0xc00e97ebd0}, 0x5e0)
	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:165 +0x77
github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotPacer).Pace(0xc008b15fc0, {0xe87b668, 0xc00e97ebd0}, 0x1d, 0x1)
	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:292 +0x14f
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*kvBatchSnapshotStrategy).Receive(_, {_, _}, _, {_, _}, {{0xe, 0x2, 0xc006dbdc70, 0xc006dbdce0, ...}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:918 +0x3239
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).receiveSnapshot(0xc00855e408, {0xe87b668, 0xc006c4bef0}, 0xc0085da008, {0x7f17175486e0, 0xc00c6f7820})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:1693 +0xcbc
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot.func1({0xe87b668, 0xc006c4bef0})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:209 +0x106
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc00b8f8b40, {0xe87b668, 0xc006c4bef0}, {0xb24c606, 0x1e}, 0xc007c52d00)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349 +0x247
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot(0xc00855e408, {0xe87b630, 0xc0071a1770}, 0xc0085da008, {0x7f17175486b8, 0xc00c6f7820})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:206 +0x1c5
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*RaftTransport).RaftSnapshot(0xc003448200, {0xe8c9450, 0xc00c6f7820})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/raft_transport.go:638 +0x4f9
github.com/cockroachdb/cockroach/pkg/kv/kvserver._MultiRaft_RaftSnapshot_Handler({0xb07aca0, 0xc003448200}, {0xe8b2960, 0xc0072393e0})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvserver/storage_services.pb.go:267 +0xe6
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.StreamServerInterceptor.func13({0xb07aca0, 0xc003448200}, {0xe8b3b50, 0xc006140870}, 0xc00969af30, 0xb677cb8)
	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:160 +0x449
google.golang.org/grpc.getChainStreamHandler.func1({0xb07aca0, 0xc003448200}, {0xe8b3b50, 0xc006140870})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0x119
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4({0xb07aca0, 0xc003448200}, {0xe8b3b50, 0xc006140870}, 0xc00969af30, 0xc006dece00)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:174 +0xd7
google.golang.org/grpc.getChainStreamHandler.func1({0xb07aca0, 0xc003448200}, {0xe8b3b50, 0xc006140870})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0x119
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor({0xc00854e000?, {{0xc006decdc0?}, {0xe8d4be0?, 0xc004c4d1f0?}}}, {0xb07aca0, 0xc003448200}, {0xe8b3b50, 0xc006140870}, 0xc00969af30, 0xc006decdc0)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:152 +0x564
google.golang.org/grpc.getChainStreamHandler.func1({0xb07aca0, 0xc003448200}, {0xe8b3b50, 0xc006140870})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0x119
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1({0xc00b8f8b40?, 0xe87b668?})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:141 +0x66
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc00b8f8b40, {0xe87b668, 0xc006c4bc80}, {0xc006783650, 0x29}, 0xc00a8af778)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349 +0x247
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2({0xb07aca0, 0xc003448200}, {0xe8b3b50, 0xc006140870}, 0xc00969af30, 0xc006decd80)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:140 +0x15c
google.golang.org/grpc.NewServer.chainStreamServerInterceptors.chainStreamInterceptors.func2({0xb07aca0, 0xc003448200}, {0xe8b3b50, 0xc006140870}, 0xc00969af30, 0xb677cb8)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1472 +0xdd
google.golang.org/grpc.(*Server).processStreamingRPC(0xc00686d2c0, {0xe8d5540, 0xc00c5a1860}, 0xc0071eb680, 0xc001c3be30, 0x12fe4180, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1636 +0x20fc
google.golang.org/grpc.(*Server).handleStream(0xc00686d2c0, {0xe8d5540, 0xc00c5a1860}, 0xc0071eb680, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1716 +0xfc5
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:947 +0x178
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 103062
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:958 +0x265
```
<details><summary>Log preceding fatal error</summary>
<p>

```
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotQueue).Admit
*   | 	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:165
*   | github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotPacer).Pace
*   | 	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:292
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*kvBatchSnapshotStrategy).Receive
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:918
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).receiveSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:1693
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot.func1
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:209
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:206
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*RaftTransport).RaftSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/raft_transport.go:638
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver._MultiRaft_RaftSnapshot_Handler
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvserver/storage_services.pb.go:267
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.StreamServerInterceptor.func13
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:160
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:174
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:152
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:141
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:140
*   | google.golang.org/grpc.NewServer.chainStreamServerInterceptors.chainStreamInterceptors.func2
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1472
*   | google.golang.org/grpc.(*Server).processStreamingRPC
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1636
*   | google.golang.org/grpc.(*Server).handleStream
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1716
*   | google.golang.org/grpc.(*Server).serveStreams.func1.1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:947
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=2</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowRangesMultipleStores.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestShowRangesMultipleStores [failed](https://tanzanite.cluster.engflow.com/invocations/default/f9978e6b-8d20-4dcd-aeaa-975a39247769?testReportRun=14&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [472ea07a5232c98536293d13bb46cca59f9f2cd0](https://github.com/cockroachdb/cockroach/commits/472ea07a5232c98536293d13bb46cca59f9f2cd0):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x185ad11]

```
Stack: 

```
goroutine 165516 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc00cf69820?, {0x8bb4548, 0xc00ec22390})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:226 +0x65
panic({0x66d4500?, 0xce28b40?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc00cf69940?, {0x8bb4548, 0xc00ec224e0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:226 +0x65
panic({0x66d4500?, 0xce28b40?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotQueue).Admit(0x0, {0x8bb4548, 0xc005262b40}, 0x565)
	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:165 +0x71
github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotPacer).Pace(0xc00cf6a720, {0x8bb4548?, 0xc005262b40?}, 0xc00e049600?, 0x48?)
	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:292 +0x6d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*kvBatchSnapshotStrategy).Receive(_, {_, _}, _, {_, _}, {{0xe, 0x2, 0xc00452de30, 0xc00452dea0, ...}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:918 +0x1f9f
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).receiveSnapshot(0xc00d464008, {0x8bb4548, 0xc00ec224e0}, 0xc007effd48, {0x7f8154aa79e0, 0xc0068c6f80})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:1693 +0x797
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot.func1({0x8bb4548, 0xc00ec224e0})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:209 +0x85
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc006dee3f0, {0x8bb4548, 0xc00ec224e0}, {0x729925b, 0x1e}, 0xc002fd1268)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349 +0x15d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot(0xc00d464008, {0x8bb4510?, 0xc00cbe0a50?}, 0xc007effd48, {0x7f8154aa79b8, 0xc0068c6f80})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:206 +0xf6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*RaftTransport).RaftSnapshot(0xc007289180, {0x8c027f0, 0xc0068c6f80})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/raft_transport.go:638 +0x30d
github.com/cockroachdb/cockroach/pkg/kv/kvserver._MultiRaft_RaftSnapshot_Handler({0x70c9720?, 0xc007289180}, {0x8bebd00, 0xc01320f880})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvserver/storage_services.pb.go:267 +0xd8
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.StreamServerInterceptor.func13({0x70c9720, 0xc007289180}, {0x8becf38, 0xc00b6225a0}, 0xc0094014e8, 0x76c2ec0)
	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:160 +0x35b
google.golang.org/grpc.getChainStreamHandler.func1({0x70c9720, 0xc007289180}, {0x8becf38, 0xc00b6225a0})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4({0x70c9720, 0xc007289180}, {0x8becf38, 0xc00b6225a0}, 0xc0094014e8?, 0xc007357b80)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:174 +0x6c
google.golang.org/grpc.getChainStreamHandler.func1({0x70c9720, 0xc007289180}, {0x8becf38, 0xc00b6225a0})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor({0xc00f500000?, {{0x470445?}, {0x8c0df00?, 0xc0104e2600?}}}, {0x70c9720, 0xc007289180}, {0x8becf38, 0xc00b6225a0}, 0xc0094014e8, 0xc007357b40)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:152 +0x384
google.golang.org/grpc.getChainStreamHandler.func1({0x70c9720, 0xc007289180}, {0x8becf38, 0xc00b6225a0})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1({0x7f8155c7ea88?, 0x8bb4548?})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:141 +0x27
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc006dee3f0, {0x8bb4548, 0xc00ec22390}, {0xc00b5f02d0, 0x29}, 0xc00789daa0)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349 +0x15d
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2({0x70c9720, 0xc007289180}, {0x8becf38, 0xc00b6225a0}, 0xc0094014e8, 0xc007357b00)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:140 +0xdf
google.golang.org/grpc.NewServer.chainStreamServerInterceptors.chainStreamInterceptors.func2({0x70c9720, 0xc007289180}, {0x8becf38, 0xc00b6225a0}, 0xc0094014e8, 0xc0068c6ee0?)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1472 +0x85
google.golang.org/grpc.(*Server).processStreamingRPC(0xc009843c20, {0x8c0e860, 0xc013a371e0}, 0xc00fc5a120, 0xc0122270b0, 0xcf25d60, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1636 +0x1247
google.golang.org/grpc.(*Server).handleStream(0xc009843c20, {0x8c0e860, 0xc013a371e0}, 0xc00fc5a120, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1716 +0x999
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:947 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 165512
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:958 +0x136
```
<details><summary>Log preceding fatal error</summary>
<p>

```
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotQueue).Admit
*   | 	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:165
*   | github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotPacer).Pace
*   | 	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:292
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*kvBatchSnapshotStrategy).Receive
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:918
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).receiveSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:1693
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot.func1
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:209
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:206
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*RaftTransport).RaftSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/raft_transport.go:638
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver._MultiRaft_RaftSnapshot_Handler
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvserver/storage_services.pb.go:267
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.StreamServerInterceptor.func13
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:160
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:174
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:152
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:141
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:140
*   | google.golang.org/grpc.NewServer.chainStreamServerInterceptors.chainStreamInterceptors.func2
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1472
*   | google.golang.org/grpc.(*Server).processStreamingRPC
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1636
*   | google.golang.org/grpc.(*Server).handleStream
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1716
*   | google.golang.org/grpc.(*Server).serveStreams.func1.1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:947
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=14</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #132991 sql: TestShowRangesMultipleStores failed [C-test-failure O-robot T-sql-queries branch-release-24.3 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowRangesMultipleStores.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestShowRangesMultipleStores [failed](https://tanzanite.cluster.engflow.com/invocations/default/b86ff27e-b525-4113-ad6c-0db94f54a0d1?testReportRun=17&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [472ea07a5232c98536293d13bb46cca59f9f2cd0](https://github.com/cockroachdb/cockroach/commits/472ea07a5232c98536293d13bb46cca59f9f2cd0):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x185ad11]

```
Stack: 

```
goroutine 165394 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc00a685820?, {0x8bb4548, 0xc010baa660})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:226 +0x65
panic({0x66d4500?, 0xce28b40?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc00a685940?, {0x8bb4548, 0xc010baa7b0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:226 +0x65
panic({0x66d4500?, 0xce28b40?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotQueue).Admit(0x0, {0x8bb4548, 0xc010baa9f0}, 0x565)
	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:165 +0x71
github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotPacer).Pace(0xc00a686720, {0x8bb4548?, 0xc010baa9f0?}, 0xc016182840?, 0x48?)
	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:292 +0x6d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*kvBatchSnapshotStrategy).Receive(_, {_, _}, _, {_, _}, {{0xe, 0x2, 0xc0032f4bd0, 0xc0032f4c40, ...}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:918 +0x1f9f
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).receiveSnapshot(0xc00ac40008, {0x8bb4548, 0xc010baa7b0}, 0xc00b3f0b48, {0x7fc9efc92dd0, 0xc00e565440})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:1693 +0x797
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot.func1({0x8bb4548, 0xc010baa7b0})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:209 +0x85
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc00d2f66c0, {0x8bb4548, 0xc010baa7b0}, {0x729925b, 0x1e}, 0xc0029ad268)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349 +0x15d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot(0xc00ac40008, {0x8bb4510?, 0xc014d89d10?}, 0xc00b3f0b48, {0x7fc9efc92da8, 0xc00e565440})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:206 +0xf6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*RaftTransport).RaftSnapshot(0xc00b2056c0, {0x8c027f0, 0xc00e565440})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/raft_transport.go:638 +0x30d
github.com/cockroachdb/cockroach/pkg/kv/kvserver._MultiRaft_RaftSnapshot_Handler({0x70c9720?, 0xc00b2056c0}, {0x8bebd00, 0xc013439260})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvserver/storage_services.pb.go:267 +0xd8
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.StreamServerInterceptor.func13({0x70c9720, 0xc00b2056c0}, {0x8becf38, 0xc013cf60f0}, 0xc015287080, 0x76c2ec0)
	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:160 +0x35b
google.golang.org/grpc.getChainStreamHandler.func1({0x70c9720, 0xc00b2056c0}, {0x8becf38, 0xc013cf60f0})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4({0x70c9720, 0xc00b2056c0}, {0x8becf38, 0xc013cf60f0}, 0xc015287080?, 0xc0146710c0)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:174 +0x6c
google.golang.org/grpc.getChainStreamHandler.func1({0x70c9720, 0xc00b2056c0}, {0x8becf38, 0xc013cf60f0})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor({0xc00bec6000?, {{0x470445?}, {0x8c0df00?, 0xc00a403620?}}}, {0x70c9720, 0xc00b2056c0}, {0x8becf38, 0xc013cf60f0}, 0xc015287080, 0xc014671080)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:152 +0x384
google.golang.org/grpc.getChainStreamHandler.func1({0x70c9720, 0xc00b2056c0}, {0x8becf38, 0xc013cf60f0})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1({0x7fc9f043ff08?, 0x8bb4548?})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:141 +0x27
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc00d2f66c0, {0x8bb4548, 0xc010baa660}, {0xc011784cf0, 0x29}, 0xc0078ceaa0)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349 +0x15d
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2({0x70c9720, 0xc00b2056c0}, {0x8becf38, 0xc013cf60f0}, 0xc015287080, 0xc014671040)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:140 +0xdf
google.golang.org/grpc.NewServer.chainStreamServerInterceptors.chainStreamInterceptors.func2({0x70c9720, 0xc00b2056c0}, {0x8becf38, 0xc013cf60f0}, 0xc015287080, 0xc00e565430?)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1472 +0x85
google.golang.org/grpc.(*Server).processStreamingRPC(0xc011973860, {0x8c0e860, 0xc00f16cb60}, 0xc019d40ea0, 0xc014e369f0, 0xcf25d60, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1636 +0x1247
google.golang.org/grpc.(*Server).handleStream(0xc011973860, {0x8c0e860, 0xc00f16cb60}, 0xc019d40ea0, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1716 +0x999
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:947 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 165375
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:958 +0x136
```
<details><summary>Log preceding fatal error</summary>
<p>

```
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotQueue).Admit
*   | 	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:165
*   | github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotPacer).Pace
*   | 	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:292
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*kvBatchSnapshotStrategy).Receive
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:918
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).receiveSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:1693
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot.func1
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:209
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:206
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*RaftTransport).RaftSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/raft_transport.go:638
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver._MultiRaft_RaftSnapshot_Handler
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvserver/storage_services.pb.go:267
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.StreamServerInterceptor.func13
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:160
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:174
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:152
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:141
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:140
*   | google.golang.org/grpc.NewServer.chainStreamServerInterceptors.chainStreamInterceptors.func2
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1472
*   | google.golang.org/grpc.(*Server).processStreamingRPC
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1636
*   | google.golang.org/grpc.(*Server).handleStream
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1716
*   | google.golang.org/grpc.(*Server).serveStreams.func1.1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:947
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=17</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #132991 sql: TestShowRangesMultipleStores failed [C-test-failure O-robot T-sql-queries branch-release-24.3 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowRangesMultipleStores.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestShowRangesMultipleStores [failed](https://tanzanite.cluster.engflow.com/invocations/default/fa614228-f8fe-4e0c-99f1-fcd3dd9c8417?testReportRun=6&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [472ea07a5232c98536293d13bb46cca59f9f2cd0](https://github.com/cockroachdb/cockroach/commits/472ea07a5232c98536293d13bb46cca59f9f2cd0):

Fatal error:

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x185ad11]

```
Stack: 

```
goroutine 153954 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc01605b820?, {0x8bb4548, 0xc01d5864b0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:226 +0x65
panic({0x66d4500?, 0xce28b40?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc01605b940?, {0x8bb4548, 0xc01d586600})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:226 +0x65
panic({0x66d4500?, 0xce28b40?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotQueue).Admit(0x0, {0x8bb4548, 0xc01d586c90}, 0x570)
	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:165 +0x71
github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotPacer).Pace(0xc01605c720, {0x8bb4548?, 0xc01d586c90?}, 0xc005f5b340?, 0x48?)
	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:292 +0x6d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*kvBatchSnapshotStrategy).Receive(_, {_, _}, _, {_, _}, {{0xe, 0x2, 0xc004f5d2d0, 0xc004f5d340, ...}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:918 +0x1f9f
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).receiveSnapshot(0xc010bb7008, {0x8bb4548, 0xc01d586600}, 0xc004717208, {0x7ffa1d0769a0, 0xc001eab210})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:1693 +0x797
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot.func1({0x8bb4548, 0xc01d586600})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:209 +0x85
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc0070b7d40, {0x8bb4548, 0xc01d586600}, {0x729925b, 0x1e}, 0xc006853268)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349 +0x15d
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot(0xc010bb7008, {0x8bb4510?, 0xc00908e870?}, 0xc004717208, {0x7ffa1d076978, 0xc001eab210})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:206 +0xf6
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*RaftTransport).RaftSnapshot(0xc010ab2700, {0x8c027f0, 0xc001eab210})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/raft_transport.go:638 +0x30d
github.com/cockroachdb/cockroach/pkg/kv/kvserver._MultiRaft_RaftSnapshot_Handler({0x70c9720?, 0xc010ab2700}, {0x8bebd00, 0xc0053fc440})
	github.com/cockroachdb/cockroach/pkg/kv/kvserver/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvserver/storage_services.pb.go:267 +0xd8
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.StreamServerInterceptor.func13({0x70c9720, 0xc010ab2700}, {0x8becf38, 0xc00bb56b40}, 0xc006faca50, 0x76c2ec0)
	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:160 +0x35b
google.golang.org/grpc.getChainStreamHandler.func1({0x70c9720, 0xc010ab2700}, {0x8becf38, 0xc00bb56b40})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4({0x70c9720, 0xc010ab2700}, {0x8becf38, 0xc00bb56b40}, 0xc006faca50?, 0xc00c7179c0)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:174 +0x6c
google.golang.org/grpc.getChainStreamHandler.func1({0x70c9720, 0xc010ab2700}, {0x8becf38, 0xc00bb56b40})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor({0xc00d218000?, {{0x470445?}, {0x8c0df00?, 0xc00ed72780?}}}, {0x70c9720, 0xc010ab2700}, {0x8becf38, 0xc00bb56b40}, 0xc006faca50, 0xc00c717980)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:152 +0x384
google.golang.org/grpc.getChainStreamHandler.func1({0x70c9720, 0xc010ab2700}, {0x8becf38, 0xc00bb56b40})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481 +0xb2
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1({0x7ffa1e5b60a0?, 0x8bb4548?})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:141 +0x27
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc0070b7d40, {0x8bb4548, 0xc01d5864b0}, {0xc010f97050, 0x29}, 0xc00dfc9aa0)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349 +0x15d
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2({0x70c9720, 0xc010ab2700}, {0x8becf38, 0xc00bb56b40}, 0xc006faca50, 0xc00c717940)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:140 +0xdf
google.golang.org/grpc.NewServer.chainStreamServerInterceptors.chainStreamInterceptors.func2({0x70c9720, 0xc010ab2700}, {0x8becf38, 0xc00bb56b40}, 0xc006faca50, 0xc001eab200?)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1472 +0x85
google.golang.org/grpc.(*Server).processStreamingRPC(0xc00d1a6f00, {0x8c0e860, 0xc005805860}, 0xc018591320, 0xc00d5a68d0, 0xcf25d60, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1636 +0x1247
google.golang.org/grpc.(*Server).handleStream(0xc00d1a6f00, {0x8c0e860, 0xc005805860}, 0xc018591320, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1716 +0x999
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:947 +0xbb
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 153915
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:958 +0x136
```
<details><summary>Log preceding fatal error</summary>
<p>

```
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotQueue).Admit
*   | 	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:165
*   | github.com/cockroachdb/cockroach/pkg/util/admission.(*SnapshotPacer).Pace
*   | 	github.com/cockroachdb/cockroach/pkg/util/admission/snapshot_queue.go:292
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*kvBatchSnapshotStrategy).Receive
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:918
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).receiveSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_snapshot.go:1693
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot.func1
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:209
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).HandleSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store_raft.go:206
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*RaftTransport).RaftSnapshot
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/raft_transport.go:638
*   | github.com/cockroachdb/cockroach/pkg/kv/kvserver._MultiRaft_RaftSnapshot_Handler
*   | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/kvserver_go_proto_/github.com/cockroachdb/cockroach/pkg/kv/kvserver/storage_services.pb.go:267
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.StreamServerInterceptor.func13
*   | 	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:160
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:174
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:152
*   | google.golang.org/grpc.getChainStreamHandler.func1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1481
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:141
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:349
*   | github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2
*   | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:140
*   | google.golang.org/grpc.NewServer.chainStreamServerInterceptors.chainStreamInterceptors.func2
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1472
*   | google.golang.org/grpc.(*Server).processStreamingRPC
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1636
*   | google.golang.org/grpc.(*Server).handleStream
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1716
*   | google.golang.org/grpc.(*Server).serveStreams.func1.1
*   | 	google.golang.org/grpc/external/org_golang_google_grpc/server.go:947
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=6</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #132991 sql: TestShowRangesMultipleStores failed [C-test-failure O-robot T-sql-queries branch-release-24.3 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowRangesMultipleStores.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

sql.TestShowRangesMultipleStores [failed](https://tanzanite.cluster.engflow.com/invocations/default/b1ce1a73-3cf0-4f83-96db-b4e555ad6751?testReportRun=1&testReportShard=14&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [546c489ae32a83b8d312977156cc8f2be93b6a49](https://github.com/cockroachdb/cockroach/commits/546c489ae32a83b8d312977156cc8f2be93b6a49):


```
=== RUN   TestShowRangesMultipleStores
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestShowRangesMultipleStores459654815
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:139: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    show_ranges_test.go:142: -- test log scope end --
test logs left over in: outputs.zip/logTestShowRangesMultipleStores459654815
--- FAIL: TestShowRangesMultipleStores (126.94s)
=== RUN   TestShowRangesMultipleStores/SHOW_RANGES_FROM_DATABASE_system_WITH_DETAILS
    sql_runner.go:116: 
        	Error Trace:	pkg/testutils/sqlutils/sql_runner.go:129
        	            				pkg/testutils/sqlutils/sql_runner.go:372
        	            				pkg/sql/show_ranges_test.go:132
        	Error:      	Received unexpected error:
        	            	query '
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:375
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	pkg/testutils/soon.go:71
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	pkg/util/retry/retry.go:213
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	pkg/testutils/soon.go:77
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:129
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:372
        	            	  | github.com/cockroachdb/cockroach/pkg/sql_test.TestShowRangesMultipleStores.func1
        	            	  | 	pkg/sql/show_ranges_test.go:132
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1689
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (2) query '
        	            	  | SELECT DISTINCT
        	            	  | 		(
        	            	  | 		array_position(replica_localities, lease_holder_locality)
        	            	  | 		= array_position(replicas, lease_holder)
        	            	  | 		)
        	            	  | 	FROM [SHOW RANGES FROM DATABASE system WITH DETAILS]': expected:
        	            	  | true
        	            	  |
        	            	  | got:
        	            	  | true
        	            	  | NULL
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
    --- FAIL: TestShowRangesMultipleStores/SHOW_RANGES_FROM_DATABASE_system_WITH_DETAILS (64.46s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=14</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #134173 sql: TestShowRangesMultipleStores failed [C-test-failure O-robot T-sql-queries branch-release-24.3 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowRangesMultipleStores.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/show_ranges_test.go
+++ b/pkg/sql/show_ranges_test.go
@@ -15,6 +15,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqltestutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/testcluster"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
@@ -80,11 +81,13 @@ FROM [SHOW RANGES FROM TABLE t WITH DETAILS]`
 	}
 }
 
-// TestRangeLocalityBasedOnNodeIDs tests that the leaseholder_locality shown in
+// TestShowRangesMultipleStores tests that the leaseholder_locality shown in
 // SHOW RANGES works correctly.
 func TestShowRangesMultipleStores(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+
+	skip.UnderRace(t, "the test is too heavy")
 
 	ctx := context.Background()
 
@@ -116,7 +119,7 @@ func TestShowRangesMultipleStores(t *testing.T) {
 	)
 	assert.NoError(t, tc.WaitForFullReplication())
 
-	// Scatter a system table so that the lease is unlike to be on node 1.
+	// Scatter a system table so that the lease is unlikely to be on node 1.
 	sqlDB := sqlutils.MakeSQLRunner(tc.Conns[0])
 	sqlDB.Exec(t, "ALTER TABLE system.jobs SCATTER")
 	// Ensure that the localities line up.
@@ -128,7 +131,7 @@ func TestShowRangesMultipleStores(t *testing.T) {
 		"SHOW RANGE FROM INDEX system.jobs@jobs_status_created_idx FOR ROW ('running', now(), 0)",
 	} {
 		t.Run(q, func(t *testing.T) {
-			// Retry because if there's not a leaseholder, you can NULL.
+			// Retry because if there's not a leaseholder, you can get a NULL.
 			sqlDB.CheckQueryResultsRetry(t,
 				fmt.Sprintf(`
 SELECT DISTINCT

--- a/pkg/testutils/sqlutils/sql_runner.go
+++ b/pkg/testutils/sqlutils/sql_runner.go
@@ -26,7 +26,7 @@ import (
 // convenience functions to run SQL statements and fail the test on any errors.
 type SQLRunner struct {
 	DB                   DBHandle
-	SucceedsSoonDuration time.Duration // defaults to testutils.DefaultSucceedsSoonDuration
+	SucceedsSoonDuration time.Duration // defaults to testutils.DefaultSucceedsSoonDuration or testutils.RaceSucceedsSoonDuration
 	MaxTxnRetries        int           // defaults to 0 for unlimited retries
 }
 
@@ -124,7 +124,7 @@ func (sr *SQLRunner) succeedsWithin(t Fataler, f func() error) {
 	helperOrNoop(t)()
 	d := sr.SucceedsSoonDuration
 	if d == 0 {
-		d = testutils.DefaultSucceedsSoonDuration
+		d = testutils.SucceedsSoonDuration()
 	}
 	require.NoError(requireT{t}, testutils.SucceedsWithinError(f, d))
 }
@@ -230,7 +230,7 @@ func (sr *SQLRunner) ExpectErrWithTimeout(
 	helperOrNoop(t)()
 	d := sr.SucceedsSoonDuration
 	if d == 0 {
-		d = testutils.DefaultSucceedsSoonDuration
+		d = testutils.SucceedsSoonDuration()
 	}
 	err := timeutil.RunWithTimeout(context.Background(), "expect-err", d, func(ctx context.Context) error {
 		_, err := sr.DB.ExecContext(ctx, query, args...)

